strawberry/types/types.py

```diff
@@ -23,6 +29,80 @@
 
     from strawberry.field import StrawberryField
 
+T = TypeVar("T")
+
+
+def _get_interfaces(cls: Type[T]) -> List[TypeDefinition]:
+    interfaces = []
+
+    for base in cls.__bases__:
+        type_definition = cast(
+            Optional[TypeDefinition], getattr(base, "__strawberry_definition__", None)
+        )
+
+        if type_definition and type_definition.is_interface:
+            interfaces.append(type_definition)
+
+        for inherited_interface in _get_interfaces(base):
+            interfaces.append(inherited_interface)
+
+    return interfaces
+
+
+def _get_fields(cls: Type) -> List["StrawberryField"]:
+    """Get all the strawberry fields off a strawberry.type cls
+
+    This function returns a list of StrawberryFields (one for each field item), while
+    also paying attention the name and typing of the field.
+
+    StrawberryFields can be defined on a strawberry.type class as either a dataclass-
+    style field or using strawberry.field as a decorator.
+
+    >>> import strawberry
+    >>> @strawberry.type
+    ... class Query:
+    ...     type_1a: int = 5
+    ...     type_1b: int = strawberry.field(...)
+    ...     type_1c: int = strawberry.field(resolver=...)
+    ...
+    ...     @strawberry.field
+    ...     def type_2(self) -> int:
+    ...         ...
+
+    Type #1:
+        A pure dataclass-style field. Will not have a StrawberryField; one will need to
+        be created in this function. Type annotation is required.
+
+    Type #2:
+        A field defined using @strawberry.field as a decorator around the resolver. The
+        resolver must be type-annotated.
+
+    The StrawberryField.python_name value will be assigned to the field's name on the
+    class if one is not set by either using an explicit strawberry.field(name=...) or by
+    passing a named function (i.e. not an anonymous lambda) to strawberry.field
+    (typically as a decorator).
+    """
+    # Deferred import to avoid import cycles
+    from strawberry.field import StrawberryField
+
+    fields: Dict[str, StrawberryField] = {}
+
+    # TODO: What is this?
```

Comment:
@patrick91 can you tell me if that is necessary? or what is that for? fields test are passing locally for me without it.